File: MovieProject/MovieProject.Web/Modules/ServerTypes/MovieDB/MovieCastColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieCastRow } from "./MovieCastRow";

export interface MovieCastColumns {
    MovieCastId: Column<MovieCastRow>;
    MovieTitle: Column<MovieCastRow>;
    PersonFirstName: Column<MovieCastRow>;
    Character: Column<MovieCastRow>;
}

export class MovieCastColumns extends ColumnsBase<MovieCastRow> {
    static readonly columnsKey = 'MovieDB.MovieCast';
    static readonly Fields = fieldsProxy<MovieCastColumns>();
}

File: MovieProject/MovieProject.Web/Modules/MovieDB/MovieCast/MovieCastColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MovieProject.MovieDB.Columns;

[ColumnsScript("MovieDB.MovieCast")]
[BasedOnRow(typeof(MovieCastRow), CheckNames = true)]
public class MovieCastColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MovieCastId { get; set; }
    public string MovieTitle { get; set; }
    public string PersonFirstName { get; set; }
    [EditLink]
    public string Character { get; set; }
}